test/cpp/tensorexpr/test_loopnest.cpp

```diff
@@ -3849,8 +3849,27 @@ static Stmt* splitTailReorder(Tensor* b) {
   LoopNest nest({b});
   auto loops = nest.getLoopStmtsFor(b);
   nest.splitWithTail(loops[0], kVectorWidth, &outer, &inner, &tail);
-  loops = nest.getLoopStmtsFor(b);
-  nest.reorderAxis(loops[1], loops[2]);
+  // Now the loopnests will look like:
+  //
+  // for (int n_outer = 0; ...
+  //   for (int n_inner = 0; ...
+  //     b[n_outer * 8 + n_inner] = float(0);
+  //     for (int m = 0; ...
+  //       b[n_outer * 8 + n_inner] = ReduceOp(...);
+  //
+  // for (int n_tail = 0; ...
+  //   b[n_tail + ((100 - 0) / 8) * 8] = float(0);
+  //   for (int m = 0; ...
+  //     b[n_tail + ((100 - 0) / 8) * 8] = ReduceOp(...);
+  //
+  // Since there are 4 writes to b, we will get 4 loopnests from the
+  // call to `getAllLoopNestsWritingToBuf` below.
+  //
+  // Write #2: "b[n_outer * 8 + n_inner] = ReduceOp(...)"
+  // Loopnest #2: {n_outer, n_inner, m};
+  // We will have to reorder n_inner and m.
+  auto loopnests = nest.getAllLoopNestsWritingToBuf(b->buf());
+  nest.reorderAxis(loopnests[1][1], loopnests[1][2]);
```

Comment:
Can you add to this test a `FileCheck` showing that the IR has the expected form?

Reply:
Done.